allenact_plugins/clip_plugin/clip_preprocessors.py

```diff
@@ -0,0 +1,160 @@
+
+    def __init__(
+        self,
+        rgb_input_uuid: str,
+        clip_model_type: str,
+        pool: bool,
+        device: Optional[torch.device] = None,
+        device_ids: Optional[List[torch.device]] = None,
+        **kwargs: Any,
+    ):
+        assert clip_model_type in clip.available_models()
+
+        if clip_model_type == "RN50":
+            output_shape = (2048, 7, 7)
+        elif clip_model_type == "RN50x16":
+            output_shape = (3072, 7, 7)
+        else:
+            raise NotImplementedError(
+                f"Currently `clip_model_type` must be one of 'RN50' or 'RN50x16'"
+            )
+
+        if pool:
+            output_shape = output_shape[:1]
+
+        self.clip_model_type = clip_model_type
+
+        self.pool = pool
+
+        self.device = torch.device("cpu") if device is None else device
+        self.device_ids = device_ids or cast(
+            List[torch.device], list(range(torch.cuda.device_count()))
+        )
+        self._resnet: Optional[ClipResNetEmbedder] = None
+
+        low = -np.inf
+        high = np.inf
+        shape = output_shape
+
+        input_uuids = [rgb_input_uuid]
+        assert (
+            len(input_uuids) == 1
+        ), "resnet preprocessor can only consume one observation type"
+
+        observation_space = gym.spaces.Box(low=low, high=high, shape=shape)
+
+        super().__init__(**prepare_locals_for_super(locals()))
+
+    @property
+    def resnet(self) -> ClipResNetEmbedder:
+        import clip
+
+        if self._resnet is None:
+            self._resnet = ClipResNetEmbedder(
+                clip.load(self.clip_model_type, device=self.device)[0], pool=self.pool
+            ).to(self.device)
+        return self._resnet
+
+    def to(self, device: torch.device) -> "ClipResNetPreprocessor":
+        self._resnet = self.resnet.to(device)
+        self.device = device
+        return self
+
+    def process(self, obs: Dict[str, Any], *args: Any, **kwargs: Any) -> Any:
+        x = obs[self.input_uuids[0]].to(self.device).permute(0, 3, 1, 2)  # bhwc -> bchw
+        # If the input is depth, repeat it across all 3 channels
+        if x.shape[1] == 1:
+            x = x.repeat(1, 3, 1, 1)
+        x = self.resnet(x).float()
+        return x
+
+
+class ClipTextPreprocessor(Preprocessor):
+
+    def __init__(
+        self,
+        goal_sensor_uuid: str,
+        object_types: List[str],
+        device: Optional[torch.device] = None,
+        device_ids: Optional[List[torch.device]] = None,
+        **kwargs: Any,
+    ):
+        try:
+            import clip
+            self.clip = clip
+        except ImportError as _:
+            raise ImportError(
+                "Cannot `import clip` when instatiating `CLIPResNetPreprocessor`."
+                " Please install clip from the openai/CLIP git repository:"
+                "\n`pip install git+https://github.com/openai/CLIP.git@3b473b0e682c091a9e53623eebc1ca1657385717`"
+            )
```

Comment:
We have an `ImportChecker` context that should help when doing things like this:
```python
with ImportChecker(
    "Cannot `import clip` when instatiating `CLIPResNetPreprocessor`.  Please install clip from the openai/CLIP git"
    " repository: \n`pip install git+https://github.com/openai/CLIP.git@3b473b0e682c091a9e53623eebc1ca1657385717`\n"
):
    import clip
```
will try to import clip and otherwise throw a `ModuleNotFoundError` exception.

Reply:
Removing this code for redundancy (clip is already imported at the top).

Also, removing the import check, because CLIP is a mandatory part of this plugin (installed in extra_environment). Is it fine to instead just expect that users will see an import error and install the plugin requirements?